nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/aws/S3DirectBatchHandler.java

```diff
@@ -0,0 +1,329 @@
+
+public class S3DirectBatchHandler extends AbstractBatchHandler {
+
+    private static final Log log = LogFactory.getLog(S3DirectBatchHandler.class);
+
+    public static final String DEFAULT_S3_HANDLER_NAME = "s3direct";
+
+    private static final Pattern REGEX_MULTIPART_ETAG = Pattern.compile("-\\d+$");
+
+    private static final long FIVE_GB = 5_368_709_120L;
+
+    private String transientStoreName;
+
+    private TransientStore transientStore;
+
+    private AWSSecurityTokenService stsClient;
+    private AmazonS3 s3Client;
+
+    private String roleArnToAssume;
+    private String awsRegion;
+    private boolean s3AccelerationSupported;
+    private String bucket;
+    private String baseBucketKey;
+
+    public S3DirectBatchHandler() {
+        super(DEFAULT_S3_HANDLER_NAME);
+    }
+
+    @Override
+    public Batch newBatch() {
+        return initBatch();
+    }
+
+    @Override
+    public Batch getBatch(String batchId) {
+        TransientStore transientStore = getTransientStore();
+        Map<String, Serializable> batchEntryParams = transientStore.getParameters(batchId);
+
+        if (batchEntryParams == null) {
+            if (!hasBatch(batchId)) {
+                return null;
+            }
+            batchEntryParams = new HashMap<>();
+        }
+
+        String batchProvider = batchEntryParams.getOrDefault("provider", getName()).toString();
+        batchEntryParams.remove("provider");
+
+        if (!getName().equalsIgnoreCase(batchProvider)) {
+            return null;
+        }
+
+        Batch batch = new Batch(transientStore, getName(), batchId, batchEntryParams, this);
+
+        Map<String, Object> batchExtraInfo = batch.getBatchExtraInfo();
+        AssumeRoleResult assumeRoleResult = stsClient.assumeRole(
+                new AssumeRoleRequest().withRoleSessionName(batch.getKey()).withRoleArn(roleArnToAssume));
+        batchExtraInfo.put("awsSecretKeyId", assumeRoleResult.getCredentials().getAccessKeyId());
+        batchExtraInfo.put("awsSecretAccessKey", assumeRoleResult.getCredentials().getSecretAccessKey());
+        batchExtraInfo.put("awsSessionToken", assumeRoleResult.getCredentials().getSessionToken());
+        batchExtraInfo.put("bucket", bucket);
+        batchExtraInfo.put("baseKey", baseBucketKey);
+        batchExtraInfo.put("expiration", assumeRoleResult.getCredentials().getExpiration().toInstant().toEpochMilli());
+        batchExtraInfo.put("region", awsRegion);
+        batchExtraInfo.put("useS3Accelerate", s3AccelerationSupported);
+
+        return batch;
+    }
+
+    private boolean hasBatch(String batchId) {
+        return !StringUtils.isEmpty(batchId) && transientStore.exists(batchId);
+    }
+
+    @Override
+    public Batch newBatch(String batchId) {
+        return initBatch(batchId);
+    }
+
+    @Override
+    protected void init(Map<String, String> configProperties) {
+        if (!containsRequired(configProperties)) {
+            throw new NuxeoException();
```

Comment:
Could you add a message to the exception explaining that configuration properties are not valid ?

Reply:
I can replace it by throwing an `IllegalArgumentException` and wrapping it in a `NuxeoException`, wdyt?

Reply:
Hum, don't really know what we use to do in such situation. Maybe a bit verbose to throw a `NuxeoException` wrapping an `IllegalArgumentException` compared to a small message. Your call.

Reply:
@efge , wdyt ?

Reply:
Rename `containsRequired` to `checkRequiredProperties`, make it return `void` and throw the `NuxeoException` itself if there's a problem.

Reply:
Should i add `NuxeoException` as Checked Exception ?

Reply:
I've modified the code a bit to avoid changing the "contracts" again. Please re-check

Reply:
Following @efge comment, you should have something like this:
```
public void checkRequiredProperties(Map<String, String> configProperties) {
        String missingProperties = Stream.of(Constants.S3TransientStoreConfig.Properties.AWS_SECRET_KEY_ID,
                Constants.S3TransientStoreConfig.Properties.AWS_SECRET_ACCESS_KEY,
                Constants.S3TransientStoreConfig.Properties.AWS_ROLE_ARN,
                Constants.S3TransientStoreConfig.Properties.AWS_REGION,
                Constants.S3TransientStoreConfig.Properties.AWS_BUCKET,
                Constants.S3TransientStoreConfig.Properties.TRANSIENT_STORE_NAME)
                                         .filter(property -> isEmpty(configProperties.get(property)))
                                         .collect(Collectors.joining(","));
        if (!missingProperties.isEmpty()) {
            throw new NuxeoException(MessageFormat.format("Properties {0} are missing", missingProperties));
        }
}
```